homie-controller/tests/integration.rs

```diff
@@ -0,0 +1,220 @@
+//! Integration test between homie-controller and homie-device crates. Starts a device and a
+//! controller connected to the same MQTT broker, and ensures that the controller can discover the
+//! device as expected.
+
+use futures::future::ready;
+use homie_controller::{Event, HomieController, State};
+use homie_device::{HomieDevice, Node, Property, SpawnError};
+use librumqttd::{Broker, Config};
+use rumqttc::{ConnectionError, MqttOptions, StateError};
+use std::io::ErrorKind;
+use std::sync::{Arc, Mutex};
+use std::thread;
+
+// A high port number which is hopefully not in use, to use for the MQTT broker.
+const PORT: u16 = 10883;
```

Comment:
if you do `while cargo test -- homie-controller/tests/integration.rs ; do true; done` how many loops through do you get before it errors out?

Reply:
... `while cargo test --test integration ; do true; done` - seems to work fine forever.

Reply:
I guess that's fine then?

Reply:
With logging enabled and running the loop in two different tabs eventually produces:
```
running 1 test
 ERROR librumqttd > Accept loop error: "I/O Address already in use (os error 48)"
test test_device ... FAILED

failures:

---- test_device stdout ----
thread 'test_device' panicked at 'called `Result::unwrap()` on an `Err` value: Request(SendError(..))', homie-controller/tests/integration.rs:53:32
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Without logging you might be a bit stuffed trying to debug that.

https://docs.rs/env_logger/0.8.2/env_logger/#capturing-logs-in-tests suggests a better way. Maybe pretty_env_logger has the same method?

Reply:
Done. That means we get the error logs back, but I think the proper fix here is to avoid logging errors when nothing is actually wrong so I'll try to fix that.